<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Calibration_model_moduleClass</class>
 <widget class="QWidget" name="Calibration_model_moduleClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>560</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Calibration_model_module</string>
  </property>
  <property name="windowIcon">
   <iconset resource="Camera_calibration_module.qrc">
    <normaloff>:/Camera_calibration_module/icon/pet.ico</normaloff>:/Camera_calibration_module/icon/pet.ico</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout_4">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_5" stretch="0,0,0">
     <item>
      <widget class="QGroupBox" name="detect_groupBox">
       <property name="title">
        <string>Detection Parameters</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QStackedWidget" name="detect_stackedWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="page">
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1">
                <item>
                 <widget class="QLabel" name="peak_threashold_label">
                  <property name="text">
                   <string>Peak Threshold:</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="peak_threashold_doubleSpinBox">
                  <property name="minimum">
                   <double>0.010000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>0.990000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.150000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QCheckBox" name="high_distortion_checkBox">
                <property name="text">
                 <string>High Distortion</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="page_2"/>
          <widget class="QWidget" name="page_3"/>
          <widget class="QWidget" name="page_4"/>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="camera_groupBox">
       <property name="title">
        <string>Camera Parameters</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QCheckBox" name="optimize_checkBox">
            <property name="text">
             <string>Optimize the calibration board morphology</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QCheckBox" name="uniform_f_checkBox">
              <property name="text">
               <string>Uniform Focal Length</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="skew_checkBox">
              <property name="text">
               <string>Skew</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1">
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <item>
               <widget class="QLabel" name="k_distort_label">
                <property name="text">
                 <string>Radial Distortion:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="p_distort_label">
                <property name="text">
                 <string>Tangential Distortion:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="s_distort_label">
                <property name="text">
                 <string>Thin Prism Distortion:</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QComboBox" name="k_distort_comboBox">
                <property name="currentIndex">
                 <number>3</number>
                </property>
                <item>
                 <property name="text">
                  <string>0</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>1</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>2</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>3</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>4</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>5</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>6</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="p_distort_comboBox">
                <property name="currentIndex">
                 <number>1</number>
                </property>
                <item>
                 <property name="text">
                  <string>0</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>2</string>
                 </property>
                </item>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="s_distort_comboBox">
                <property name="currentIndex">
                 <number>0</number>
                </property>
                <item>
                 <property name="text">
                  <string>0</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>2</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>4</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="1,0,0,1,0,0,1">
          <item>
           <widget class="QCheckBox" name="verify_checkBox">
            <property name="text">
             <string>Verify the stability</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="de_sample_label">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Desampling ratio (%):</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="desample_spinBox">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="maximum">
             <number>80</number>
            </property>
            <property name="value">
             <number>30</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="desample_label">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Number</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="desample_number_label">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>999</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Optimization parameter</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="thread_label">
              <property name="text">
               <string>Number of threads:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="timeout_label">
              <property name="text">
               <string>Timeout (seconds):</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="max_iter_label">
              <property name="text">
               <string>Maximum number of Iterations:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="stop_label">
              <property name="text">
               <string>Gradient for stop:</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <widget class="QLabel" name="loss_label">
                <property name="text">
                 <string>Loss type:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="comboBox">
                <property name="currentIndex">
                 <number>3</number>
                </property>
                <item>
                 <property name="text">
                  <string>Trivial</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Huber</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>SoftLOne</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Cauchy</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ArctanLoss</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_7">
            <item>
             <widget class="QSpinBox" name="thread_spinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>128</number>
              </property>
              <property name="value">
               <number>6</number>
              </property>
              <property name="displayIntegerBase">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="timeout_spinBox">
              <property name="minimum">
               <number>60</number>
              </property>
              <property name="maximum">
               <number>172800</number>
              </property>
              <property name="value">
               <number>3600</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="max_iter_spinBox">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>10000</number>
              </property>
              <property name="value">
               <number>1000</number>
              </property>
              <property name="displayIntegerBase">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <item>
               <widget class="QSpinBox" name="stop_spinBox"/>
              </item>
              <item>
               <widget class="QLabel" name="stop_label_2">
                <property name="text">
                 <string>E</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="stop_spinBox_2">
                <property name="minimum">
                 <number>-20</number>
                </property>
                <property name="maximum">
                 <number>1</number>
                </property>
                <property name="value">
                 <number>-10</number>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="loss_doubleSpinBox">
              <property name="value">
               <double>1.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="Camera_calibration_module.qrc"/>
 </resources>
 <connections/>
</ui>
